float64 x
float64 y
float64 z
float64 theta
float64 v
float64 vz
float64 t
